src/entities/TextNode/index.spec.ts

```diff
@@ -1,14 +1,154 @@
-import { describe, it, expect } from '@jest/globals';
+import { describe, it, expect, beforeEach } from '@jest/globals';
 import { TextNode } from './index';
-import type { BlockNode } from '../BlockNode';
+import { createFormattingNodeName, FormattingNode } from '../FormattingNode';
 
 describe('TextNode', () => {
-  it('should create an instance', () => {
-    const node = new TextNode({
-      value: 'test',
-      parent: {} as BlockNode,
+  const initialText = 'initial text';
+  const text = 'some text';
+  let node: TextNode;
+
+  beforeEach(() => {
+    node = new TextNode({
+      value: initialText,
+    });
+  });
```

Comment:
its better to create node inside each case:

1. test arrange could be different
2. redundant class constructor will be called when test creates own node
3. Code coverage could be messy

Reply:
https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

Reply:
DRY is an eventually bad practise. It often makes architecture dirtier, less readable, creating redundant complication and abstraction. 

I've provided arguments, why dry is not a best solution here. Follow the Unit test best practice, especially the AAA pattern:

https://github.com/goldbergyoni/javascript-testing-best-practices#section-1-the-test-anatomy-1